diff.py

```diff
@@ -2050,9 +2078,21 @@ def normalize_stack(row: str, arch: ArchSettings) -> str:
     return re.sub(arch.re_sprel, "addr(sp)", row)
 
 
-def imm_matches_everything(row: str, arch: ArchSettings) -> bool:
-    # (this should probably be arch-specific)
-    return "(." in row
+def imm_matches_everything(field: str, arch: ArchSettings) -> bool:
+    if arch.name == "ppc":
+        if "...data" in field:
+            return True
+
+        field = "".join(field.rsplit("@ha", 1))
+        field = "".join(field.rsplit("@l", 1))
+        field = "".join(field.rsplit("@sda21", 1))
+
+        return re.search(re.compile(r"\A@\d+\Z"), field) != None
+    else:
+        return (
+            re.search(re.compile(r"\A%(hi|lo)\(\.[a-z]+(\+0x([a-f]|\d)+)?\)"), field)
```

Comment:
this should be done just for mips, and the regex pre-compiled instead of compiled once per call

`([a-f]|\d)` -> `[a-f0-9]`

(imo the existing `"(." in row` is also just fine to keep, along with a comment saying what it targets. It has the advantage of less complexity)

Reply:
aha, I changed it away from "(." in row, because you asked me too. Atleast that is what I understood